test/parallel/test-worker-process-argv.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { Worker, isMainThread } = require('worker_threads');
+
+if (isMainThread) {
+  assert.throws(() => {
+    new Worker(__filename, { argv: 'foo' });
+  }, {
+    code: 'ERR_INVALID_ARG_TYPE'
+  });
+
+  new Worker(__filename, { argv: [null, 'foo', 123, Symbol('bar')] })
+    .on('exit', common.mustCall((code) => {
+      assert.strictEqual(code, 0);
+    }));
+
+  new Worker(`
+    const assert = require('assert');
+    assert.deepStrictEqual(
+      process.argv,
+      [process.execPath, 'null', 'foo', '123', String(Symbol('bar'))]
```

Comment:
I know this matches `node --eval`, but … this feels a bit wrong to me.

Shouldn’t the second entry always be the filename for the current script, even if it’s a fake one because of `eval: true`?

Reply:
Would it make sense that the filename is `[worker eval]` for `node -p '__filename'` printing `[eval]`?

Reply:
Yeah, I’m good with that personally :+1:

Reply:
It's more common to get filename from the `__filename` right? Since `__filename` in eval worker already presents and has a value of `[worker eval]`, is it required to add the filename to argv while we don't generally have it in our `node -e 'process.argv'`?

Reply:
> It's more common to get filename from the `__filename` right?

`__filename` and `process.argv[1]` are different things; the former is the path of the current CJS module, the latter is the path of the *main* module.

> Since `__filename` in eval worker already presents and has a value of `[worker eval]`, is it required to add the filename to argv while we don't generally have it in our `node -e 'process.argv'`?

You can feel free to omit this, but I’d consider it inconsistent that `process.argv` *doesn’t* include a fake filename for `node -e`. It seems to break expectations for me, because in `node -e` the structure of `process.argv` should be the same as elsewhere.

Changing that for `node -e` is a breaking change – and I’d be willing to open a PR for that – but here we have the chance to be consistent from the beginning.

Reply:
Tested against bash, in which argv never includes filename, in both eval mode or script mode. So it might make sense to add a fake filename in eval mode since we have it in script mode.

I'll update the PR to include the filename in eval worker's argv to keep consistency with script mode.